code/main.py

```diff
@@ -0,0 +1,221 @@
+#     for angle 0 - 2 pi
+#         for translate_x 0 - imag_max
+#             for translate_y 0 - imag_max
+#                 transform = transform(angle, translate)
+#                 add transformed imag into try_mask
+
+import os
+import numpy as np
+import imageio
+from skimage.transform import resize
+from skimage.color import rgb2gray
+from matplotlib import pyplot as plt
+from scipy.ndimage.morphology import binary_fill_holes
+from skimage.morphology import binary_closing
+from skimage.color import label2rgb
+from skimage.feature import canny
+from skimage import measure
+import cv2 as cv
+import random as rng
+
+## Rotation of countours start
+def cart2pol(x, y):
+    theta = np.arctan2(y, x)
+    rho = np.hypot(x, y)
+    return theta, rho
+
+
+def pol2cart(theta, rho):
+    x = rho * np.cos(theta)
+    y = rho * np.sin(theta)
+    return x, y
+
+
+def rotate_contour(cnt, angle, center, img_size):
+    M = cv.moments(cnt)
+    cx = int(M['m10']/M['m00'])
+    cy = int(M['m01']/M['m00'])
+
+    cnt_norm = cnt - [cx, cy]
+
+    coordinates = cnt_norm[:, 0, :]
+    xs, ys = coordinates[:, 0], coordinates[:, 1]
+    thetas, rhos = cart2pol(xs, ys)
+
+    thetas = np.rad2deg(thetas)
+    thetas = (thetas + angle) % 360
+    thetas = np.deg2rad(thetas)
+
+    xs, ys = pol2cart(thetas, rhos)
+
+    cnt_norm[:, 0, 0] = xs
+    cnt_norm[:, 0, 1] = ys
+
+    cnt_rotated = cnt_norm + [img_size[0] / 2, img_size[1] / 2]
+    cnt_rotated = cnt_rotated.astype(np.int32)
+
+    return cnt_rotated
+# Rotation of countours
+
+
+ref_objects = []
+ref_bin_objects = []
+
+def load_ref_objects(path, is_low = False):
+    global ref_objects
+    if is_low and os.path.isdir(path + "/low"):
+        ref_objects = [imageio.imread(path + "/low/" + fname) for fname in os.listdir(path+"/low") if fname.endswith(".jpg")]
+        return
+    ref_objects = [imageio.imread(path + "/" + fname) for fname in os.listdir(path) if fname.endswith(".jpg")]
+    ref_objects = ref_objects[1:12]
+    if is_low:
+        ref_objects = [resize(t, (256, 256 * t.shape[1] / t.shape[0])) for t in ref_objects]
+        ref_objects = [t[0 : t.shape[0] - 0, 0 : t.shape[1] - 0] for t in ref_objects]
+        os.mkdir(path+"/low")
+        for i in range(len(ref_objects)):
+            imageio.imwrite(path + "/low/" + str(i) + ".jpg",ref_objects[i])
+
+def cv2_image_from_imageio(img):
+    return (img * 255).astype(np.uint8)
+
+def binarize(image):
+    gray = rgb2gray(image)
+    my_edge_map = binary_closing(canny(gray, sigma=0.4), selem=np.ones((3, 3)))
+    my_edge_segmentation = binary_fill_holes(my_edge_map)
+
+    return cv2_image_from_imageio(my_edge_segmentation)
+
+
+def check_place(placement: np.ndarray):
+    err_cnt = 0
+    for el in placement.flat:
+        if el > 255:
+            err_cnt += 1
+            if err_cnt > placement.shape[0] * placement.shape[1] / 5000:
+                return False
+    return True
+
+
+def try_place(destination: np.ndarray, objects, idx):
```

Comment:
You have a very clear function name. But it would be nice if there was a short description:)